### PR TITLE
Naming

### PR DESCRIPTION
**Pull request by skaae**
- Adds multinomial cost  (please check :))
- rename `mu` to `mean` across all classes. This matches numpy. 
- use 'log_var' in all classes
- Fix typo in confusion matrix batchaddmasked method
- some pep8 changes.

### CODE REVIEW

File: parmesan/distributions.py [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+c = - 0.5 * math.log(2*math.pi)
 
 
-c = - 0.5 * math.log(2*math.pi)
 def log_normal(x, mean, sd):
     """
     Compute log pdf of a Gaussian distribution with diagonal covariance, at values x.
 
-        .. math:: \log p(x) = \log \mathcal{N}(x; \mu, \sigma^2I)
+        .. math:: \log p(x) = \log \mathcal{N}(x; \mean, \sigma^2I)
```
Comment (wuaalb) [inline parmesan/distributions.py]: The `\mu` in the docstrings was supposed to be a LaTeX symbol.

I'd either leave it like it was or change to something like
```
    .. math:: \log p(x) = \log \mathcal{N}(x; \mathtt{mean}, \exp \mathtt{log_var} I)
```

Comment (skaae) [inline parmesan/distributions.py]: My  mistake. I just refactored. I'll fix that. 

I'm not sure what the best naming is. I think we should match numpy but i don't really like `sigma` 
from http://docs.scipy.org/doc/numpy/reference/generated/numpy.random.lognormal.html#numpy.random.lognormal

**Comment by casperkaae** [comment]
I agree that this is a better naming. 

I'll Merge unless you have further changes?

**Comment by skaae** [comment]
unrelated note: What linewidth do you use? I think we should try to stick with 80

File: parmesan/distributions.py [hunk 2]
```diff
@@ -86,7 +88,34 @@ def log_bernoulli(x, p):
     """
     return -T.nnet.binary_crossentropy(p, x)
 
-def kl_normal2_stdnormal(mean, logvar):
+
+def log_multinomial(x, p):
+    """
+    Compute log pdf of a multinomial distribution
+
+        .. math:: \log p(x; p) = \log \sum_x p(x) log(q(x))
```
Comment (wuaalb) [inline parmesan/distributions.py]: Should be `.. math:: \log p(x; p) = \sum_x p(x) \log q(x)`?

File: parmesan/distributions.py [hunk 3]
```diff
@@ -86,7 +88,34 @@ def log_bernoulli(x, p):
...
+    probability.
+
+
+    Parameters
+    ----------
+    x : Theano tensor
+        Values at which to evaluate pdf. Either an integer vector or a
+        samples by class matrix with class probabilities.
+    p : Theano tensor
+        samles by class matrix with predicted calss probabilities
```
Comment (wuaalb) [inline parmesan/distributions.py]: Couple typos here..

**Comment by skaae** [comment]
@wuaalb: Thanks. I was too sloppy :)

squased the commits

File: parmesan/distributions.py [hunk 4]
```diff
@@ -86,7 +88,34 @@ def log_bernoulli(x, p):
     """
     return -T.nnet.binary_crossentropy(p, x)
 
-def kl_normal2_stdnormal(mean, logvar):
+
+def log_multinomial(x, p):
+    """
+    Compute log pdf of multinomial distribution
+
+        .. math:: \log p(x; p) = \log \sum_x p(x) \log q(x)
```
Comment (wuaalb) [inline parmesan/distributions.py]: Still a `\log` too many imho :)

Comment (skaae) [inline parmesan/distributions.py]: yes. I'm apparently unable to copy paste from Theano's help page.

**Comment by casperkaae** [comment]
Looks good :) Any further changes otherwise i'll merge later tonight ?

**Comment by skaae** [comment]
While we are at various fixes: Maybe we should reverse the order in the confusion matrix such that it correpsonds to the cost functions i.e `some_function(model_pred, true_value)`. The confusion matrix has the order reversed. 

https://github.com/casperkaae/parmesan/blob/master/parmesan/utils.py#L56

**Comment by casperkaae** [comment]
I'll merge this one now and fix the confusion matrix together with #29

**Pull request merged**